python/cudf/cudf/core/indexed_frame.py

```diff
@@ -867,3 +867,66 @@ def resample(
             if isinstance(self, cudf.Series)
             else cudf.core.resample.DataFrameResampler(self, by=by)
         )
+
+    def _reset_index(self, level, drop, col_level=0, col_fill=""):
+        """Shared path for DataFrame.reset_index and Series.reset_index."""
+        if col_level != 0:
+            raise NotImplementedError(
+                "col_level parameter is not supported yet."
+            )
+
+        if col_fill != "":
+            raise NotImplementedError(
+                "col_fill parameter is not supported yet."
+            )
+
+        if level is not None and not isinstance(level, (tuple, list)):
+            level = (level,)
+        _check_duplicate_level_names(level, self._index.names)
+
+        # Split the columns in the index into data and index columns
+        (
+            data_columns,
+            index_columns,
+            column_names,
+            index_names,
+        ) = self._index._split_columns_by_levels(level)
+        if index_columns:
+            index = _index_from_data(
+                dict(zip(range(len(index_columns)), index_columns)),
+                name=self._index.name,
+            )
+            if isinstance(index, MultiIndex):
+                index.names = index_names
+            else:
+                index.name = index_names[0]
+        else:
+            index = RangeIndex(len(self))
+
+        if drop:
+            return self._data, index
+
+        new_column_data = {}
+        for name, col in zip(column_names, data_columns):
+            if name == "index" and "index" in self._data:
+                name = "level_0"
+            new_column_data[name] = col
+        return {**new_column_data, **self._data}, index
```

Comment:
Just double-checking, I assume this ordering is important for consistency with pandas?

Reply:
Yes, the only place in pandas doc that mentions this is in the return section of `Series.reset_index`:

> The newly created columns will come first in the DataFrame, followed by the original Series values.

Reply:
Could you add a comment here just so that it's easier to keep track without having to look at our tests to verify that ordering?

Reply:
done